weave

```diff
@@ -741,8 +752,11 @@ setup_cni() {
     if install_cni_plugin $CNI_PLUGIN_DIR ; then
         upgrade_cni_plugin $CNI_PLUGIN_DIR
     fi
-    if [ -d $HOST_ROOT/etc/cni/net.d -a ! -f $HOST_ROOT/etc/cni/net.d/10-weave.conf ] ; then
-        create_cni_config $HOST_ROOT/etc/cni/net.d/10-weave.conf $HAIRPIN_MODE
+    if [ -d $HOST_ROOT/etc/cni/net.d -a -f $HOST_ROOT/etc/cni/net.d/10-weave.conf ] ; then
```

Comment:
First one is intended to delete if there is `10-weave.conf`. Like in case of upgrades. So both fresh installs and existing clusters after upgrade will only have `10-weave.conflist`. Please let me know if you see any issue with this approach.

Reply:
Yes, I get what you are trying to do, that part is fine.
My point is that `-f $HOST_ROOT/etc/cni/net.d/10-weave.conf` implies `-d $HOST_ROOT/etc/cni/net.d`, so you don't need both parts of the `-a`.

Reply:
Make sense. I removed redundant check.